### ISSUE TITLE
NPM installs multiple nested copies of a module

### ISSUE THREAD

**Issue by tusharmath** [npm/npm#11381]
I am using webpack to build a single JS file. Here is the structure of the dependency tree.
```
node_modules/
---lodash@3
---depA
------node_modules
---------lodash@4
---depB
------node_modules
---------lodash@4
---depC
------node_modules
---------lodash@4
```

As you can see — My module depends on lodash@3 but because the other dependencies are dependent on lodash@4 I end up bloating the build with multiple copies of the same version (version 4) of lodash is being used.

Attaching the [stats.json.zip](https://github.com/npm/npm/files/113922/stats.json.zip) to be used on — https://chrisbateman.github.io/webpack-visualizer/

May be this is by design, what should I do to not have multiple copies of lodash version 4. I don't want to upgrate lodash on my main module since that could break a lot of things.

**Comment by KenanY** [comment]
@tusharmath `npm --version`? Maybe `npm dedupe` will help?

**Comment by tusharmath** [comment]
NPM Version: 3.3.12

**Comment by tusharmath** [comment]
No dedupe doesn't work at all.

**Comment by ashleygwilliams** [comment]
this is the intended behavior. for more explanation, see: https://docs.npmjs.com/how-npm-works/npm3

tl;dr: since `lodash@3` occupies the top level of `node_modules`, no other version of `lodash` can, and, therefore, if another version of `lodash` is required for a dependency, it will be nested.

**Comment by tusharmath** [comment]
@ashleygwilliams I am sure it is, but hear me out —
It creates a lot problems when using with browserify or webpack because multiple copies of the library is loaded in the browser which causes weird behaviour in the application. The only solution I have in mind right now is to create a proxy package which depends on `lodash@3` and exports it as is. This temporarily solves the problem but is obviously a hack.  

How do I mitigate this particular problem?
